processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -608,12 +612,11 @@ private void reportErrorForUnmappedTargetPropertiesIfRequired() {
             // fetch settings from element to implement
             ReportingPolicyPrism unmappedTargetPolicy = getUnmappedTargetPolicy();
 
-            //we handle forged methods differently than the usual source ones. in
-            if ( method instanceof ForgedMethod ) {
+            //we handle automapping forged methods differently than the usual source ones. in
+            if ( method instanceof ForgedMethod && ( (ForgedMethod) method ).isAutoMapping() ) {
```

Comment:
New method on `ForgedMethod` or on `Method`? 

`isForgeNameBased` would be similar to the `isAutoMapping` right?